CleverRaven/Cataclysm-DDA
# Add splash screen to Object Creator

#### Summary
Features "Add spash screen to Object Creator"


#### Purpose of change
Show to the user that the application is loading. Before this change nothing would show up until the application is fully loaded.


#### Describe the solution
It's a grey rectangle with black text in it that says 'Initializing Object Creator' and it is visible while the application is starting. There is a second splash screen that shows when the main window is showing. It disappears when the application is loaded. It's not much but it's better then nothing.
Removing 1 of the splashscreens will mean that the user sees nothing instead of that splash screen. The two splash screens cannot overlap that is why we need both of them.

Also, as part of the troubleshooting I changed the declaration of the itemgroup window and the spell window to be in line with the mod selection window. 


#### Describe alternatives you've considered
Show a image splash screen instead of a plain text one. I even had an image ready:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/50166150/13bb244d-7e7e-4795-a193-5171375da633)

The problem is that I don't know where to put it on the filesystem. The /object_creator/ folder is not there in the release version of the game. All the other folders do not seem an appropriate place to put it.

Another alternative is a loading bar or at least a % completed, but I lost the tutorial for it. 



#### Testing
On both Windows and Linux:
- Start the object_creator
- See splash screen:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/50166150/b6cd608d-da3c-4eff-99c4-53ef09af71ae)

- See the second splash screen before the main menu loads:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/50166150/3cd65593-ebe6-4a40-a74d-df803df6df7a)





#### Additional context
